lib/dry/schema/dsl.rb

```diff
@@ -84,6 +103,30 @@ def set_type(name, spec)
         types[name] = resolve_type(spec).meta(omittable: true)
       end
 
+      def key_map(types = self.types)
+        keys = types.keys.each_with_object([]) { |a, e|
+          e << key_value(a, types[a])
+        }
+        km = KeyMap.new(keys)
+
+        if hash_type === :symbolized
```

Comment:
Shouln't it be `==`, same at line 124

Reply:
If this is indeed a bug, perhaps we need to add a spec that would have picked this up?